extension-manifest-v3/src/background/adblocker.js

```diff
@@ -70,18 +72,48 @@ async function reloadEngine(enabledEngines) {
     );
 
     console.info(
-      `[adblocker] engine reloaded with: ${enabledEngines.join(', ')}`,
+      `[adblocker] Main engine reloaded with: ${enabledEngines.join(', ')}`,
     );
   } else {
     engines.create(engines.MAIN_ENGINE);
-    console.info('[adblocker] Engine reloaded with no filters');
+    console.info('[adblocker] Main engine reloaded with no filters');
   }
 }
 
-engines.addChangeListener(engines.CUSTOM_ENGINE, async () => {
-  reloadEngine(getEnabledEngines(options));
-});
+engines.addChangeListener(engines.CUSTOM_ENGINE, reloadMainEngine);
+
+let updating = false;
+async function updateEngines() {
+  if (updating) return;
+
+  try {
+    updating = true;
+    const enabledEngines = getEnabledEngines(options);
+
+    if (enabledEngines.length) {
+      let updated = false;
+      // Update engines from the list of enabled engines
+      for (const id of enabledEngines) {
+        if (id === engines.CUSTOM_ENGINE) continue;
+        updated = (await engines.update(id).catch(() => false)) || updated;
+      }
+
+      // Reload the main engine after all engines are updated
+      if (updated) await reloadMainEngine();
 
+      // Update TrackerDB engine
+      trackerdb.setup.pending && (await trackerdb.setup.pending);
+      await engines.update(engines.TRACKERDB_ENGINE).catch(() => null);
```

Comment:
shouldn't trackerdb update even if no other engine is enabled?

Reply:
I made it on purpose - I think after all, it's best to all only onboarded users to update the trackerdb more often than a release - those not onboarded will have an update with a new release. It also simplifies the case by setting the timestamp only when engines are enabled. Without that, it must be set not to update trackerdb repeatedly if the user doesn't want to onboard.

Reply:
okay, wanted to be sure that's the case